models/validation-inference-scripts/hammah-models/hammah-inference.py

```diff
@@ -40,82 +44,86 @@ def date2min(time):
         return int(timesince.total_seconds() // 60)
 
     form = "%Y-%m-%dT%H:%M:%SZ"
+
     def stript(s):
         obj = datetime.datetime.strptime(s, form)
         return obj
 
     def back_to_string(obj):
         return "{}-{}:{}:{}".format(f"{obj.month:02}", f"{obj.day:02}", f"{obj.hour:02}", f"{obj.minute:02}")
-    
-    
-    
-    
-    
-    
-    
-    
 
-    X_val=pd.read_csv(validationdata)
-
-
-    col_list=['userIdentityaccountId','eventSource','eventName','sourceIPAddress','userAgent','userIdentitytype','apiVersion', 'userIdentityprincipalId', 'userIdentityarn', 'userIdentityaccessKeyId', 'tlsDetailsclientProvidedHostHeader', 'errorCode','errorMessage','userIdentitysessionContextsessionIssueruserName']
+    X_val = pd.read_csv(validationdata)
+
+    col_list = [
+        'userIdentityaccountId',
+        'eventSource',
+        'eventName',
+        'sourceIPAddress',
+        'userAgent',
+        'userIdentitytype',
+        'apiVersion',
+        'userIdentityprincipalId',
+        'userIdentityarn',
+        'userIdentityaccessKeyId',
+        'tlsDetailsclientProvidedHostHeader',
+        'errorCode',
+        'errorMessage',
+        'userIdentitysessionContextsessionIssueruserName'
+    ]
 
     for i in list(X_val):
         if i not in col_list:
-            X_val=X_val.drop(i,axis=1)
+            X_val = X_val.drop(i, axis=1)
 
     with open(model, 'rb') as f:
-         model = dill.load(f)
+        model = dill.load(f)
 
     scores = model.get_anomaly_score(X_val)
     X_val['ae_anomaly_score'] = scores
 
     X_val.sort_values('ae_anomaly_score', ascending=False).head(10)
-    #since inference is done, add the original columns back so the output will be the same as the input format
-    #X_val['ts_anomaly']=X_val_original['ts_anomaly']
+    # since inference is done, add the original columns back so the output will be the same as the input format
+    # X_val['ts_anomaly']=X_val_original['ts_anomaly']
     df = cudf.read_csv("hammah-user123-validation-data.csv")
     df = df.sort_values(by=['eventTime'])
     timearr = df.eventTime.to_array()
     START = stript(timearr[0])
-    END = stript(timearr[-1])
     timeobj = list(map(stript, timearr))
     hs = list(map(date2min, timeobj))
-    n, bins = cp.histogram(cp.array(hs), bins = cp.arange(0, max(hs)))
+    n, _ = cp.histogram(cp.array(hs), bins=cp.arange(0, max(hs)))
```

Comment:
Maybe a nitpick, but I'd like to see all these stacked lines broken out a bit into blocks of logical steps or sub functions.